lib/src/components/NxSegmentedButton/NxSegmentedButton.tsx

```diff
@@ -30,29 +30,23 @@ const NxSegmentedButton = forwardRef<HTMLDivElement, Props>(
             isOpen,
             type,
             onToggleOpen: externalOnToggleCollapse,
-            onKeyDown: externalOnKeyDown,
-            onCloseClick,
-            onCloseKeyDown,
             ...attrs
           } = props,
           classes = classnames('nx-segmented-btn', className, {
             'nx-segmented-btn--open': isOpen
           }),
           wrappedMenuItems = React.Children.map<ReactElement, ReactElement>(children, item => (
             <NxOverflowTooltip>{item}</NxOverflowTooltip>
-          )),
-          { onKeyDown, onToggleCollapse, menuRef, toggleRef, onMenuClosing } =
-              useDropdownEvents(isOpen, disabled, externalOnToggleCollapse, onCloseClick, onCloseKeyDown,
-                  externalOnKeyDown);
+          ));
 
-      return (
-        <div ref={ref} className={classes} onKeyDown={onKeyDown} { ...attrs }>
+      const toggleElement: AbstractDropdownToggleElement = ({ toggleRef, onToggleCollapse }) => (
+        <>
           <NxButton type={type || undefined}
                     variant={variant}
                     className="nx-segmented-btn__main-btn"
                     onClick={onClick}
                     disabled={disabled || undefined}>
-            {buttonContent}
+            { buttonContent }
```

Comment:
This isn't generally how we format this

Reply:
> This isn't generally how we format this

Fixed: a3e9a119a3904ddcecaa7ab82bcc38bcc6ccfcd3